synapse/storage/databases/main/event_push_actions.py

```diff
@@ -567,7 +682,8 @@ def _get_notif_unread_count_for_user_room(
             WHERE user_id = ?
                AND room_id = ?
                AND ea.stream_ordering > ?
-               {clause}
+               {stream_ordering_clause}
+               {thread_id_clause}
```

Comment:
Can we inline this clause?

Does our SQL stuff play nicely with `None` -> `NULL` in the SQL?

Reply:
We can't inline this because this is dealing with unthreaded (`thread_id=None` vs threaded (`thread_id=xxx`), not main timeline (`thread_id=MAIN_TIMELINE`) vs thread (`thread_id=xxx`).

But it would be nice to add a comment to the clause above explaining the nuance here

Reply:
> Does our SQL stuff play nicely with `None` -> `NULL` comparison in the SQL?

`NULL` is generally not equal to other `NULL`. I don't think there's an easy way to inline this.

This is not dealing with unthreaded vs. threaded things -- it is deciding if you want to get information on a single thread or *all* threads.